lib/compiler/ops-to-jsir.ts

```diff
@@ -825,6 +825,20 @@ export default class OpCompiler {
         });
     }
 
+    private _compileVerification(verification : PointWiseOp.Verification) {
+        const answer = this._compileFilter(compileBooleanExpressionToOp(verification.booleanExpression), this._currentScope);
+        this._irBuilder.add(new JSIr.SetKey(getRegister('$output', this._currentScope), '__verification_answer', answer));
+
+        this._currentScope.set('__verification_answer', {
+            type: 'scalar',
+            register: answer,
+            tt_type: Type.Boolean,
+            direction: 'output',
+            isInVarScopeNames: true
```

Comment:
Can you monitor() a boolean question? What are semantics?
(`isInVarScopeNames` controls whether a field affects monitors or not)

Reply:
I misunderstood the meaning of this field then, why the name? also I don't think there is a comment about this confusing name...

Reply:
Yeah I think the name is terrible and it's partly historical, and I don't really remember the rationale for the name originally. I think originally we had a thing called "varScope" which was a record with all the variables in scope, and "varScopeNames" which was the list of names, used for monitoring. Then we got a boolean indicating whether something was in "varScopeNames", and eventually "varScopeNames" went away but the bad boolean remained.

I think the boolean properly means "is this an output parameter of the current ThingTalk expression" - as opposed to some special compiler-internal variable like $output, a let-assignment, or the input parameter of an enclosing procedure. As it happens, that is only really useful for monitoring though.